pico/tools/src/main/java/io/helidon/pico/tools/TypeTools.java

```diff
@@ -1153,7 +1028,7 @@ static String extractInjectionPointTypeInfo(TypeSignature sig,
                             || isListType(typeArgClassName)
                             || typeArgClassName.equals(Collections.class.getName())) {
                         // not handled
-                    } else if (isProviderType(typeArgClassName)) {
+                    } else if (isProviderType(TypeName.create(typeArgClassName))) {
```

Comment:
would prefer to avoid creation during an if - not performant at all

Reply:
the performance problem is exactly where? This is an annotation processor, not a server.

Reply:
I am creating the type name earlier, and using its methods to check for list and collection.